kong/plugins/hmac-auth/access.lua

```diff
@@ -29,6 +31,63 @@ local SIGNATURE_NOT_SAME = "HMAC signature does not match"
 
 local _M = {}
 
+local hmac = {
+  ["hmac-sha1"] = function(secret, data)
+    return crypto.hmac.digest("sha1", data, secret, true)
+  end,
+  ["hmac-sha256"] = function(secret, data)
+    return crypto.hmac.digest("sha256", data, secret, true)
+  end,
+  ["hmac-sha384"] = function(secret, data)
+    return crypto.hmac.digest("sha384", data, secret, true)
+  end,
+  ["hmac-sha512"] = function(secret, data)
+    return crypto.hmac.digest("sha512", data, secret, true)
+  end
+}
+
+local function list_as_set(list)
+  local set = {}
```

Comment:
since we know the size of this table, a minor optimization here would be to use `table.new` to create the appropriately-sized table.